File: AtCorder2/A26.java

import java.util.*;
import java.util.stream.*;
import java.io.*;

// ABC
public class Main {
  static boolean isPrime(int number) {
        if (number == 2) {
            return true;
        }
        if (number < 2 || number % 2 == 0) {
            return false;
        }
        int limit = (int) Math.sqrt(number);
        for (int counter = 3; counter <= limit; counter += 2) {
            if (number % counter == 0) {
                return false;
            }
        }
        return true;
  }
  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    final int Q = Integer.parseInt(br.readLine().trim());
    for (int i = 0; i < Q; i++) {
      int x = Integer.parseInt(br.readLine().trim());
      if (isPrime(x)) {
        System.out.println("Yes");
      } else {
        System.out.println("No");
      }
    }
    br.close();
  }
}

File: AtCorder2/A26.py
# 鉄則問題集A26（素数判定問題）
import sys
import math


def solve():
    input_data = iter(map(int,sys.stdin.read().split()))
    n = next(input_data)
    for _ in range(n):
        x = next(input_data)
        print("Yes" if is_prime(x) else "No")

def is_prime(number):
    if number <= 1:
        return False
    if number == 2:
        return True
    if number % 2 == 0:
        return False
    for i in range(3, int(math.sqrt(number)) + 1, 2):
        if number % i == 0:
            return False
    return True


if __name__ == "__main__":
    solve()